rpc/http.go

```diff
@@ -324,6 +324,19 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	connInfo.HTTP.Host = r.Host
 	connInfo.HTTP.Origin = r.Header.Get("Origin")
 	connInfo.HTTP.UserAgent = r.Header.Get("User-Agent")
+	connInfo.HTTP.Header = r.Header
+
+	// Read the body and make it available in the request scope. The request's
+	// Body ReaderCloser is replaced to read over this same slice.
+	body, err := io.ReadAll(io.LimitReader(r.Body, int64(s.httpBodyLimit)))
+	if err != nil {
+		// `httpBodyLimit` is already checked when validated
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	connInfo.HTTP.Body = body
+	r.Body = io.NopCloser(bytes.NewReader(body))
```

Comment:
Looping back here I agree that implementing this at the rpc server level doesn't make sense.

However the performance hit isn't a big deal for the proxy since it's low throughput. Looking through how get implements the auth out of the server impl via the [http handlers](https://github.com/ethereum-optimism/op-geth/blob/optimism/node/rpcstack.go#L316), we can do something similar as well.

Going to close this PR as no op-geth changes will be needed anymore. It would nice of go-ethereum refactored some of the handler code to work natively with the server